packages/gitbook/src/components/DocumentView/Expandable/Details.tsx

```diff
@@ -0,0 +1,78 @@
+'use client'
+
+import { useParams } from 'next/navigation';
+import React from 'react';
+
+import { ClassValue, tcls } from '@/lib/tailwind';
+
+
+/**
+ * Details component rendered on client so it can expand dependent on url hash changes.
+ */
+export function Details(props: { children : React.ReactNode; id: string; contentIds?: string[]; open?: boolean; className?: ClassValue; }) {
```

Comment:
Instead of accumulating the `contentIds` and potentially build a big array that needs to be passed client side.

I think the logic should be:

```
const hash = useHash()

React.useEffect(() => {
  if (hash) {
     const descendant = document.querySelectorAll(`#${hash}`); // carefully you might need to handle encoding a better approach here ex with getById
     if (descendant) {
        // open it
     }
  }
}, [hash]);

```

Reply:
Another approach could be to have one single hook in the layout that everytime the hash changes, lookup all the ancestors of the DOM element and set open to them if they are `<detail>`, it'll not be react-ish but more DOM-ish API

Reply:
Thanks for the pointers! I definitely wasn't very confident in my solution performance wise. Will take another look.